libusb/os/windows_winusb.c

```diff
@@ -523,6 +523,9 @@ static int windows_assign_endpoints(struct libusb_device_handle *dev_handle, uin
 		return r;
 	}
 
+	if (iface >= conf_desc->bNumInterfaces) {
+		return LIBUSB_ERROR_NOT_FOUND;
```

Comment:
Should we maybe print an error message here as well?

Reply:
Feel free to suggest a line of code, I don't know the debugging convention in libusb. I only did it for the darwin error because there was an example right next to my patch.

Reply:
I think this should be fine:
`usbi_err(HANDLE_CTX(dev_handle), "interface out of range for device");`
or even:
`usbi_err(HANDLE_CTX(dev_handle), "interface %d out of range for device", iface);`
I don't know if we could hint stronger to this being a broken device but it would be better to check for and report about that during enumeration, so I'd leave it like this.

Reply:
ok, I added the interface number to darwin and added the same message to windows

Reply:
Excellent.